Fixes #165, added missing docs (#227)
MozillaReality/ecsy: site/docs/manual/Architecture.md

world = new World();
```

> The `World` constructor accept an option object with the following parameters:

Reviewer: ```suggestion
The `World` constructor accepts an option object with the following parameters:
```

- ***entityClass***: Provide the base class for entities that implements or extends `Entity`.

> - ***entityPoolSize***: Define the initial entity pool size for entities. It can help to avoid GC during execution if the application keep extending the pool dynamically at execution time.

Reviewer: ```suggestion
- ***entityPoolSize***: Define the initial entity pool size for entities. It can help to avoid GC during execution if the application expands the pool dynamically at execution time.
```

```javascript

> // We know we will have around 10k enemies in our game initially so let's reserve that size initially instead of keep extending the pool as we need more entities

Reviewer: ```suggestion
// We know we will initially have around 10k enemies in our game so let's allocate 10k enemies initially and expand the pool as needed.
```

If there is a `reactive query` (A query that *listens* for entities added or removed to it or which components has changed, [more info](/manual/Architecture?id=reactive-queries)) on the list of queries defined by a system, this system is called `reactive system` as it will react to changes on the entities and its components.

> It's important to notice that if you plan to mutate the results of a query while you are iterating it (eg: adding or removing components that will make not match the query structure anymore, or removing the entity itself) you should traverse the results in reverse order:

Reviewer: ```suggestion
If you plan to mutate the results of a query while you are iterating it (eg: adding or removing components that will not match the query structure anymore, or removing the entity itself) you should traverse the results in reverse order:
```

```javascript
let results = this.queries.queryA.results;
for (var i = 0; i < results.length; i++) {
  let entity = results[i];
  if (i === 1) {

>     // This will cause the results list to mutate and results.length will be decremented and you won't reach the end elements in there.

Reviewer: ```suggestion
    // This will cause the results list to be mutated, results.length will be decremented and you won't reach the end elements.
```

    entity.remove();
  }
}

// The correct way to do it
let results = this.queries.queryA.results;

> for (var i = 0; i < results.length; i++) {

Reviewer: Shouldn't we be decrementing here?

Author: Actually yes ;D that's the main point actualy of this hehe changing it

// The correct way to do it
let results = this.queries.queryA.results;

> for (var i = results.length - 1; i >= 0; i++) {

Reviewer 2: `i++` should be `i--` for this to decrement.

Because of that is important that you define an appropriate execution order based on the needs for your reactive systems.

There is one special use case when removing components and entities. When using `System State Components` they should be removed explicitly and they will not get removed if `entity.remove` is being called. [More info](/manual/Architecture?id=system-state-components)

## Extending core functionality

It is possible to provide a custom `Entity` class to modify the default behaviour.

> To do so you need to import `_Entity` from `ecsy` and extends it in your class definition:

Reviewer: ```suggestion
To do so you need to import `_Entity` from `ecsy` and extend it in your class definition:
```
